lib/jxl/render_pipeline/render_pipeline.cc

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+#include "lib/jxl/render_pipeline/render_pipeline.h"
+
+#include <algorithm>
+
+#include "lib/jxl/render_pipeline/low_memory_render_pipeline.h"
+#include "lib/jxl/render_pipeline/simple_render_pipeline.h"
+#include "lib/jxl/sanitizers.h"
+
+namespace jxl {
+
+void RenderPipeline::Builder::AddStage(
+    std::unique_ptr<RenderPipelineStage> stage) {
+  stages_.push_back(std::move(stage));
+}
+
+std::unique_ptr<RenderPipeline> RenderPipeline::Builder::Finalize(
+    FrameDimensions frame_dimensions) && {
+#if JXL_ENABLE_ASSERT
+  // Check channel shifts.
+  for (const auto& stage : stages_) {
+    std::pair<size_t, size_t> current_shift = {-1, -1};
+    for (size_t c = 0; c < channel_shifts_.size(); c++) {
+      if (stage->GetChannelMode(c) == RenderPipelineChannelMode::kInOut) {
+        if (current_shift.first == -1UL) current_shift = channel_shifts_[c];
+        JXL_ASSERT(current_shift == channel_shifts_[c]);
+        channel_shifts_[c].first -= stage->settings_.shift_x;
+        channel_shifts_[c].second -= stage->settings_.shift_y;
+      }
+    }
+  }
+  for (const auto& cc : channel_shifts_) {
+    JXL_ASSERT(cc.first == 0);
+    JXL_ASSERT(cc.second == 0);
+  }
+  // Check that the last stage is not an kInOut stage for any channel, and that
+  // there is at least one stage.
+  JXL_ASSERT(!stages_.empty());
+  for (size_t c = 0; c < channel_shifts_.size(); c++) {
+    JXL_ASSERT(stages_.back()->GetChannelMode(c) !=
+               RenderPipelineChannelMode::kInOut);
+  }
+#endif
+
+  std::unique_ptr<RenderPipeline> res;
+  if (use_simple_implementation_) {
+    res = make_unique<SimpleRenderPipeline>();
```

Comment:
Maybe:

```suggestion
    res = jxl::make_unique<SimpleRenderPipeline>();
```

to be consistent with the rest of the codebase?